src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -126,6 +130,12 @@ func (r *Cluster) Default() {
 	if r.Spec.LicenseRef != nil && r.Spec.LicenseRef.Key == "" {
 		r.Spec.LicenseRef.Key = DefaultLicenseSecretKey
 	}
+
+	for i := range r.Spec.Configuration.KafkaAPI {
+		if r.Spec.Configuration.KafkaAPI[i].AuthenticationMethod == "" {
+			r.Spec.Configuration.KafkaAPI[i].AuthenticationMethod = noneAuthorizationMechanism
```

Comment:
There's some difficulty here.  The configuration makes this field optional. See: https://github.com/redpanda-data/redpanda/commit/1c9155a987b378593da732a44294ca939b3b523e#diff-37dfa5396e57820190992de850ada07cfb605311413bc76aa25d112eccc106d3R469-R495.

Leaving the authentication mechanism as unspecified means:
```
if kafka_enable_authorization is null:
   sasl if enable_sasl else none
```